src/language-js/printer-estree.js

```diff
@@ -5128,6 +5129,60 @@ function printMemberChain(path, options, print) {
   ]);
 }
 
+/**
+ * @param {import('estree').Node} node
+ * @returns {boolean}
+ */
+function isSimple(node, depth = 0) {
+  if (depth >= 2) {
+    return false;
+  }
+  const isChildSimple = child => isSimple(child, depth + 1);
+  if (
+    !node ||
+    node.type === "Literal" ||
+    node.type === "BooleanLiteral" ||
+    node.type === "NullLiteral" ||
+    node.type === "NumericLiteral" ||
+    node.type === "StringLiteral" ||
+    node.type === "Identifier" ||
+    node.type === "ThisExpression" ||
+    node.type === "Super" ||
+    node.type === "BigIntLiteral" ||
+    node.type === "PrivateName" ||
+    node.type === "ArgumentPlaceholder" ||
+    node.type === "RegExpLiteral"
+  ) {
+    return true;
+  }
+  if (node.type === "TemplateLiteral") {
+    return node.expressions.every(isChildSimple);
+  }
+  if (node.type === "ObjectExpression") {
+    return node.properties.every(p => p.shorthand);
+  }
+  if (node.type === "ArrayExpression") {
+    return node.elements.every(isChildSimple);
+  }
+  if (
+    node.type === "CallExpression" ||
+    node.type === "OptionalCallExpression" ||
+    node.type === "NewExpression"
+  ) {
+    return isSimple(node.callee) && node.arguments.every(isChildSimple);
+  }
+  if (
+    node.type === "MemberExpression" ||
+    node.type === "OptionalMemberExpression"
+  ) {
+    return isSimple(node.object) && isSimple(node.property);
```

Comment:
Again, this doesn't look good. `a(aaa.bbb(ccc.ddd(eee.fff()))` - every dot here is a `MemberExpression` and it would reset `depth` to 0, so the entire expression would pass the check.

Reply:
**Prettier pr-6685**
[Playground link](https://deploy-preview-6685--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAPAdAQwBQEp0BGe6Y2OORpAJsQA773qP24A6UIANCBLTAJbQAzslCYATuIgB3AAoSEIlJgBuEflS4gC4zGADWcGAGVae-lADmyGOICucbgAsYAWwA2AdSf94Qs2Bwxoq+-Cq+AJ7I4EIi3BZCcOIwsrqWrpjIAGaY7oncAFZCqABCugZGxpiucAAyFnDZufkgRajGFpbucACKdhDwTXmOIGbiieLRBJgEcO5atOIWMJ4aME7IABwADNyLEImeurTRi3ATKo3cAI798Km8SiCYQgC0UHBwVF9a4nC3-D+qUw6UySByw24iVc-Bs9hGQk63T6A0a4OaIxgM1WVHWyAATNxbJh+O5OgBhCCuDLRKDQK4gOyJAAqMyUEJaKgcAEkoN9YMYwEs+ABBPnGGARbpDRIAX1lQA)
```sh
--parser babel
```

**Input:**
```jsx
x.a().b().c(a(a(b(c(d().p).p).p).p))

```

**Output:**
```jsx
x.a().b().c(a(a(b(c(d().p).p).p).p));

```